webapp/src/js/controllers/inbox.js

```diff
@@ -377,12 +389,14 @@ var _ = require('underscore'),
     });
 
     $scope.unreadCount = {};
-    UnreadRecords(function(err, data) {
-      if (err) {
-        return $log.error('Error fetching read status', err);
-      }
-      $scope.unreadCount = data;
-    });
+    const initUnreadCount = () => {
+      UnreadRecords(function(err, data) {
```

Comment:
You should return here? Or multiple parts of the promise chain in `initSecondaryServices` will run in parallel (I assume you are trying to avoid that?)

Reply:
As with the XmlForms service, UnreadRecords registers a callback so it can listen for changes. Some aspects will end up running in parallel - I was trying to reduce them where possible, but it was difficult here.